lib/commonBindings.js

```diff
@@ -51,12 +53,68 @@ function guessType(attribute, device) {
     }
 
     if (attribute === constants.TIMESTAMP_ATTRIBUTE) {
-        return constants.TIMESTAMP_TYPE;
+        if (iotAgentLib.configModule.checkNgsi2()) {
+            return constants.TIMESTAMP_TYPE_NGSI2;
+        }
+        else {
+            return constants.TIMESTAMP_TYPE;
+        }
     } else {
         return constants.DEFAULT_ATTRIBUTE_TYPE;
     }
 }
 
+/**
+ * Converts the value to the corresponding type given by attribute parameter.
+ * null otherwise.
+ *
+ * @param {String} attribute        Name of the attribute to find.
+ * @param {String} value            Value of the attribute to convert.
+ * @param {Object} device           Device object containing all the information about a device.
+ * @return {Object}                 Attribute's value represented in the appropriate format.
+ */
+function convertValue(attribute, value, device) {
```

Comment:
In that case, this would be considered a new feature and should be mentioned in CNR file, eg:

```
- Add: casting from UL2.0 to JSON native format (depending on provisioned attribute types) when NGSIv2 is enabled (#233)
```

In addition, it should be added to the documentation a description of this casting feature. Maybe the best place is within https://github.com/telefonicaid/iotagent-ul/blob/master/docs/usermanual.md#ultralight-20-protocol, just after https://github.com/telefonicaid/iotagent-ul/blob/master/docs/usermanual.md#bidirectionality-syntax sub-section. (This is only just a suggestion, maybe you have a better idea on where to place it :)

(Note that in the case of IOTA-JSON documentation I don't see the need of documentation, as JSON is also the native format in the southbound. But in this case there a "strint to number, boolean, etc." feature that needs to be described).

Reply:
Fixed here: 114995a4c77ae419503b96613e62297e0ce9e79e